### PR TITLE
Bump golang:1.22.3-alpine3.18 to golang:1.22.4-alpine3.20

### PR DESCRIPTION
**Pull request by zzy2210**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

The Go version used for the image
**Description of change**

I modified the compilation script so that go version and go tag can be passed together as parameters instead of a fixed value. They are all defined in the. go-version file

Based on the make images, it runs successfully
make images

**Which issue this PR fixes**

#5214

### CODE REVIEW

File: .go-version [hunk 1]
```diff
@@ -1 +1 @@
-1.22.3
+1.22.4--alpine3.20
```
Comment (MarcosDY) [inline .go-version:1]: This file is used by people using goenv, to switch versions, can you revert this and only update alpine in dockerfile?
```suggestion
1.22.3
```

Comment (zzy2210) [reply .go-version:1]: I'm sorry, I've never used Goenv before and didn't know that.
Can I change the version in the file to 1.22.4 and update Alpine in Dockerfile?

Comment (azdagron) [inline .go-version:1]: This change needs to be reverted.

Comment (zzy2210) [reply .go-version]: Sorry, I am used to using git add * which caused me to not submit this 
Now it has been modified to 1.22.4

**Comment by nweisenauer-sap** [comment]
Hey, I think this is an important PR. I know the versions here have not been bumped before because of an incompatibility between sqlite3 and alpine 3.19, however the issue seems to be fixed now:
https://github.com/mattn/go-sqlite3/pull/1177
It was an issue on the sqlite3 side, not alpine. However, SPIRE is already using the latest version of go-sqlite3 which contains the fix.

Can we directly go to go version 1.22.5?

**Review by MarcosDY — APPROVED** [review]

**Comment by MarcosDY** [comment]
There is a DCO issue , can you resolve that? 
you may need to rebase before doing that to prevent changing another commits signatures)

**Comment by zzy2210** [comment]
> There is a DCO issue , can you resolve that?存在DCO问题，您能解决吗？ you may need to rebase before doing that to prevent changing another commits signatures)在执行此操作之前，您可能需要重新设置基础，以防止更改另一个提交签名）

I seem to have messed up. I re signed the commit through rebase, but it resulted in many commits on my PR that did not belong to me. Is this a problem? May I ask how I should solve this problem? Or should I reopen a PR to avoid this issue? 😢

**Comment by nweisenauer-sap** [comment]
Since your change is quite small, I think it's easier to raise a new PR with a single signed commit instead of going through the hassle to clean-up this one :)

**Comment by MarcosDY** [comment]
agree if you create a new PR we can merge it sooner, or you can just rebase this one, as you prefer

**Comment by zzy2210** [comment]
> agree if you create a new PR we can merge it sooner, or you can just rebase this one, as you prefer, but DCO is still failing

I have initiated a new PR  #5361 and the unique commit for this PR has been signed. I believe there will be no issues this time：）

**Comment by zzy2210** [comment]
> Since your change is quite small, I think it's easier to raise a new PR with a single signed commit instead of going through the hassle to clean-up this one :)

You're right, I chose to reopen a PR. 😃 
As for whether to choose a higher version of Go, I think it should be discussed in an issue rather than in PR

**Comment by azdagron** [comment]
Thanks @zzy2210! I've merged the other PR. We are bumping the goversion in #5362.

**Pull request closed without merging**